logsmanagement/db_api.c

```diff
@@ -0,0 +1,1096 @@
+		buffer_increase(p_query_params->results_buff, temp_msg.text_size);
+	    if(!p_query_params->keyword || !*p_query_params->keyword || !strcmp(p_query_params->keyword, " ")){
+			// TODO: decompress_text does not handle or return any errors currently. How should any errors be handled?
+			decompress_text(&temp_msg, &p_query_params->results_buff->buffer[p_query_params->results_buff->len]);
+	    	p_query_params->results_buff->len += temp_msg.text_size;
+			p_query_params->results_buff->len--; // get rid of '\0'
+			// Watch out! Changing the next line will break web_client_api_request_v1_logsmanagement()! 
+			buffer_strcat(p_query_params->results_buff, "\", \t0],\n");
+	    } 
+	    else {
+	    	decompress_text(&temp_msg, NULL);
+			size_t res_size = 0;
+	    	const int matches = search_keyword(temp_msg.data, temp_msg.text_size, 
+											   &p_query_params->results_buff->buffer[p_query_params->results_buff->len], 
+											   &res_size, 
+											   p_query_params->keyword, NULL, 
+											   p_query_params->ignore_case);
+			freez(temp_msg.data);
+
+			if(likely(matches > 0)) {
+				m_assert(res_size > 0, "res_size can't be <= 0");
+				p_query_params->results_buff->len += res_size;
+				p_query_params->results_buff->len--; // get rid of '\0'
+				// buffer_strcat(p_query_params->results_buff, "\"],\n");
+				// Watch out! Changing the next line will break web_client_api_request_v1_logsmanagement()! 
+				buffer_sprintf(p_query_params->results_buff, "\", \t%d],\n", matches);
+				p_query_params->keyword_matches += matches;
+			}
+			else if(unlikely(matches == 0)){
+				m_assert(res_size == 0, "res_size must be == 0");
+				/* No keyword matches, undo timestamp buffer_sprintf() */
+				p_query_params->results_buff->len = old_results_len;
+			}
+			else { 
+				/* matches < 0 - error during keyword search */
+				freez(temp_msg.text_compressed);
+				break;
+			}
+	    }
+		
+	    freez(temp_msg.text_compressed);
+        
+        if(p_query_params->results_buff->len >= p_query_params->quota){
+        	p_query_params->end_timestamp = temp_msg.timestamp;
+        	break;
+        }
+
+        rc = sqlite3_step(stmt_retrieve_log_msg_metadata);
+        // debug(D_LOGS_MANAG, "Query: %s\n rc:%d\n", sqlite3_expanded_sql(stmt_retrieve_log_msg_metadata), rc);
+        if (unlikely(rc != SQLITE_ROW && rc != SQLITE_DONE)) fatal_sqlite3_err(rc, __LINE__);
+    }
+
+    rc = sqlite3_exec(p_file_info->db, "END TRANSACTION;", NULL, NULL, NULL);
+    if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+    rc = sqlite3_finalize(stmt_retrieve_log_msg_metadata);
+    if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+}
+
+#define TMP_VIEW_TABLE		   "compound_view"
+#define TMP_VIEW_QUERY_PREFIX  "CREATE TEMP VIEW " TMP_VIEW_TABLE " AS SELECT * FROM (SELECT * FROM '0'."\
+							    LOGS_TABLE " INNER JOIN (VALUES(0)) ORDER BY Timestamp) "
+#define TMP_VIEW_QUERY_BODY_1  "UNION ALL SELECT * FROM (SELECT * FROM '"
```

Comment:
`SELECT *` slows down any database, please, list fields avoid overload the database.